src/column.rs

```diff
@@ -278,18 +296,25 @@ impl Column {
 				}
 			};
 
-			if let Some((table, sub_index, existing_tier, existing_address)) = existing {
+			if let Some((table, sub_index, existing_tier, existing_address)) = existing_index {
 				let existing_tier = existing_tier as usize;
 				if self.collect_stats {
+					// [Review] seems rather costy for the unsized value. Could consider storing size for
+					// those multipart value.
```

Comment:
`collect_stats` is currently enabled by default, but the plan is to disabled it once the database is stable and all the values are tweaked. So this should not be an issue.